### PR TITLE
release-1.15: Reduce minimum go toolchain in go.mod.

### LINKED ISSUE #8397: Do not set minimum go toolchain version in release branches.
**Describe the problem/challenge you have**

User requests that go.mod go and toolchain remain at patch 0. ie. `go 1.22.0` because they do not have available in their downstream system a new enough go to compile and build `1.22.8`.

[The Go toolchain refuses to load a module or workspace that declares a minimum required Go version greater than the toolchain’s own version.](https://go.dev/doc/toolchain#select:~:text=For%20example%2C%20a,go%201.22%20line.)

**Describe the solution you'd like**

The ask is that golang pinning like in https://github.com/vmware-tanzu/velero/commit/7cd92a5aac33c54b3e979775bd45e142615f8320 should be limited to outside of `go.mod` files.

We will still be able to fix CVEs by bumping toolchains used to newer go even with go.mod `go 1.22.0`

> [When GOTOOLCHAIN is set to local, the go command always runs the bundled Go toolchain.](https://go.dev/doc/toolchain#select:~:text=When%20GOTOOLCHAIN%20is%20set%20to%20local%2C%20the%20go%20command%20always%20runs%20the%20bundled%20Go%20toolchain.)

> [If the $GOROOT/go.env file is missing or does not set a default, the go command assumes GOTOOLCHAIN=local.](https://go.dev/doc/toolchain#select:~:text=If%20the%20%24GOROOT/go.env%20file%20is%20missing%20or%20does%20not%20set%20a%20default%2C%20the%20go%20command%20assumes%20GOTOOLCHAIN%3Dlocal.)

**Anything else you would like to add:**

Being fixed in https://github.com/vmware-tanzu/velero/pull/8399

Slack: https://kubernetes.slack.com/archives/C6VCGP4MT/p1731439622318149?thread_ts=1730991728.424729&cid=C6VCGP4MT

**Environment:**

- Velero version (use `velero version`):
- Kubernetes version (use `kubectl version`):
- Kubernetes installer & version:
- Cloud provider or hardware configuration:
- OS (e.g. from `/etc/os-release`):

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

### PR DESCRIPTION
**Pull request by kaovilai**
TL;DR: this change keeps 1.22.8 as go toolchain for this repo without enforcing it as minimum version on others.
<details><summary>Verification (click me!) prior to #8398 review</summary>
<p>

```
make container && docker run --rm velero/velero:main /velero version --client-only
[+] Building 60.0s (18/18) FINISHED                                                                                                        docker-container:colima-multiplat
 => [internal] load build definition from Dockerfile                                                                                                                    0.0s
 => => transferring dockerfile: 2.37kB                                                                                                                                  0.0s
 => [internal] load metadata for docker.io/paketobuildpacks/run-jammy-tiny:0.2.52                                                                                       3.6s
 => [internal] load metadata for docker.io/library/golang:1.22.8-bookworm                                                                                               3.4s
 => [auth] paketobuildpacks/run-jammy-tiny:pull token for registry-1.docker.io                                                                                          0.0s
 => [auth] library/golang:pull token for registry-1.docker.io                                                                                                           0.0s
 => [internal] load .dockerignore                                                                                                                                       0.0s
 => => transferring context: 60B                                                                                                                                        0.0s
 => CACHED [stage-2 1/3] FROM docker.io/paketobuildpacks/run-jammy-tiny:0.2.52@sha256:402b925a81a4c6985438fd37d0b22022ca688e528bbd46a38831a3702067cced                  0.0s
 => => resolve docker.io/paketobuildpacks/run-jammy-tiny:0.2.52@sha256:402b925a81a4c6985438fd37d0b22022ca688e528bbd46a38831a3702067cced                                 0.0s
 => CACHED [restic-builder 1/3] FROM docker.io/library/golang:1.22.8-bookworm@sha256:3f0457a0a56a926d93c2baf4cf0057a645e8ff69ff31314080fcc62389643b8e                   0.0s
 => => resolve docker.io/library/golang:1.22.8-bookworm@sha256:3f0457a0a56a926d93c2baf4cf0057a645e8ff69ff31314080fcc62389643b8e                                         0.0s
 => [internal] load build context                                                                                                                                       0.2s
 => => transferring context: 883.86kB                                                                                                                                   0.2s
 => CACHED [velero-builder 2/4] WORKDIR /go/src/github.com/vmware-tanzu/velero                                                                                          0.0s
 => [velero-builder 3/4] COPY . /go/src/github.com/vmware-tanzu/velero                                                                                                  1.6s
 => [restic-builder 2/3] COPY . /go/src/github.com/vmware-tanzu/velero                                                                                                  1.6s
 => [velero-builder 4/4] RUN mkdir -p /output/usr/bin &&     export GOARM=$( echo "" | cut -c2-) &&     go build -o /output/velero     -ldflags "-X github.com/vmware  49.7s
 => [restic-builder 3/3] RUN mkdir -p /output/usr/bin &&     export GOARM=$(echo "" | cut -c2-) &&     /go/src/github.com/vmware-tanzu/velero/hack/build-restic.sh &&  34.6s
 => [stage-2 2/3] COPY --from=velero-builder /output /                                                                                                                  0.2s
 => [stage-2 3/3] COPY --from=restic-builder /output /                                                                                                                  0.1s
 => exporting to docker image format                                                                                                                                    4.1s
 => => exporting layers                                                                                                                                                 2.7s
 => => exporting manifest sha256:5387d1a59b10970f9d0800ba960c4eb0f008e0d2bba5a9498b009ac478e5f4d4                                                                       0.0s
 => => exporting config sha256:f29cee5131777c8dc47571fc475f1abb2984ff1657b51c3a720c8aa453b195f2                                                                         0.0s
 => => sending tarball                                                                                                                                                  1.4s
 => importing to docker                                                                                                                                                 0.5s
 => => loading layer 4e2166c05ab9 491.52kB / 45.94MB                                                                                                                    0.5s
 => => loading layer dd8fc23784d7 98.30kB / 8.82MB                                                                                                                      0.1s

 1 warning found (use --debug to expand):
 - UndefinedVar: Usage of undefined variable '$GOPROXY' (line 58)
container: velero/velero:main
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Client:
        Version: main
        Git commit: 303d239819259b54ecd264eef176b9dc55db3871-dirty
        Go version: go1.22.8
```

</p>
</details> 

Thank you for contributing to Velero!

# Please add a summary of your change
Reduce minimum go toolchain + add verification from #8398 

Update: verification was removed per review in #8398, but we did validated this pr earlier that the correct toolchain was applied.

# Does your change fix a particular issue?

Fixes #8397

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file (`make new-changelog`)](https://velero.io/docs/main/code-standards/#adding-a-changelog) or comment `/kind changelog-not-required` on this PR.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

File: pkg/cmd/cli/version/version.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ func printVersion(w io.Writer, clientOnly bool, kbClient kbclient.Client, server
 	fmt.Fprintln(w, "Client:")
 	fmt.Fprintf(w, "\tVersion: %s\n", buildinfo.Version)
 	fmt.Fprintf(w, "\tGit commit: %s\n", buildinfo.FormattedGitSHA())
+	fmt.Fprintf(w, "\tGo version: %s\n", buildinfo.GoVersion())
```
Comment (kaovilai) [inline pkg/cmd/cli/version/version.go:72]: Is this clear enough that it is not user's local go? do we want to say `Built by Go Version` instead?

Comment (blackpiglet) [reply pkg/cmd/cli/version/version.go:72]: IMO, the information is good enough. Binary print information should already imply it's related to the binary built environment. It's not about the environment running it.

**Review by kaovilai — COMMENTED** [review]
Removed Go Version from CLI/logs per community meeting.

**Review by blackpiglet — APPROVED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,12 @@
 module github.com/vmware-tanzu/velero
 
-go 1.22.8
+// Do not pin patch version here. Leave patch at X.Y.0
+// Unset GOTOOLCHAIN to assume GOTOOLCHAIN=local where go cli version in path is used.
+// Use env GOTOOLCHAIN=auto to allow go to decide whichever is newer from go.mod or cli in path.
```
Comment (Lyndon-Li) [inline go.mod:5]: How does this change affect the release process? E.g., release of 1.15.1.
Do we need to set `GOTOOLCHAIN` anywhere during the release?

Comment (kaovilai) [reply go.mod:5]: Nope.. no need to set GOTOOLCHAIN. Per prior versions of this PR which verified the golang version used to build, it inherits local `go version` by default.

Comment (kaovilai) [reply go.mod:5]: > [In the standard configuration, the go command uses its own bundled toolchain when that toolchain is at least as new as the go or toolchain lines in the main module or workspace.
](https://go.dev/doc/toolchain#:~:text=In%20the%20standard%20configuration%2C%20the%20go%20command%20uses%20its%20own%20bundled%20toolchain%20when%20that%20toolchain%20is%20at%20least%20as%20new%20as%20the%20go%20or%20toolchain%20lines%20in%20the%20main%20module%20or%20workspace.)

**Review by shubham-pampattiwar — APPROVED** [review]
